### ISSUE TITLE
lint error if the async "color" of a function changes across subclasses

### ISSUE THREAD

**Issue by NorthIsUp** [microsoft/pyright#10031]
I don't know if this describes it well, but I'd love a lint error for if a subclass changes the aio typing of a method.

```py
class Foo:
    async def run(self) -> None:
        await asyncio.sleep(1)
        return


class Bar(Foo):
    def run(self) -> None:  # <- lint error, changes from awaitable to callable
        return None
```

and vice versa, where the superclass is def whatever and the subclass is async def whatever

thanks!

**Comment by bschnurr** [comment]
I think you need to be in typchecking mode 'standard' or override 'reportIncompatibleMethodOverride': 'error'  see https://github.com/microsoft/pyright/blob/main/docs/configuration.md#diagnostic-settings-defaults

**Comment by NorthIsUp** [comment]
appreciate that pointer, setting `reportIncompatibleVariableOverride = "error"` does detect the issue.

It would be nice as a feature to have `reportIncompatibleMethodCoroutineOverride` (naming is hard!) as its own setting. Messing up `def` vs `async def` while subclassing is often catastrophic where adding extra `**kwargs` is commonplace in python.

**Comment by erictraut** [comment]
The `reportIncompatibleVariableOverride` check is on by default if you enable standard type checking.

It's unlikely that we'd add such a specific diagnostic check — especially when it would be redundant with the `reportIncompatibleVariableOverride` check and would have backward compatibility implications for pyright users if we were to dedup them.
